game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -197,13 +357,44 @@ public double getAverageBattleRoundsFought() {
     return count / (double) results.size();
   }
 
+  /** Returns the standard deviation of the number of battle rounds. */
+  public double getAverageBattleRoundsFoughtStandardDeviation() {
+    if (results.isEmpty()) {
+      return 0.0;
+    }
+    final double mu = getAverageBattleRoundsFought();
+    final double count =
+        results.stream()
+            .mapToInt(BattleResults::getBattleRoundsFought)
+            .mapToDouble(x -> (x - mu) * (x - mu))
+            .sum();
+    if (count == 0) {
+      // If this is a 'fake' aggregate result, return 0.0
+      return 0.0;
+    }
+    return count / (double) results.size();
+  }
+
   double getDrawPercent() {
     if (results.isEmpty()) {
       return 0.0;
     }
     return results.stream().filter(BattleResults::draw).count() / (double) results.size();
   }
 
+  /** Returns the standard deviation of the draw percentage. */
+  public double getDrawStandardDeviation() {
+    if (results.isEmpty()) {
+      return 0.0;
+    }
+    // Drawing is a Bernoulli Experiment with propability p = getDrawPercent(),
```

Comment:
This seems repetitious, a good candidate to extra to a lib.

Reply:
Do you mean outsourcing the line `return p*(p-1)`? The important (and hard) part here is to realise that (in this case) drawing is a Bernoulli Experiment. This observation has to be redone in all three cases (win, lose, draw) and IMHO cannot be outsourced into a lib.  The remaining line `p*(p-1)` is so small that it does not seem worth it having a function
`getExpectedValueForBernoulliExperiment(double p)` (or `new BinomialDistribution(p).getExpectedValue()`).

Reply:
Overall it should be simpler extracted to a method. The method name can state "computeBernoulliExperimentExpectedValue(..)", which then removes the need for the comment at the site of usage. Secondarily the logic and documentation becomes consolidated. Last, it allows for you to test that component in isolation if you were to choose to do so (and arguably should).